tests/native_api_tests/test_api_model.cpp

```diff
@@ -192,57 +228,45 @@ TEST_CASE("API Model") {
     Model model_dummy{std::move(model)};
     model = std::move(model_dummy);
 
-    SUBCASE("Simple power flow") {
-        model.calculate(options, single_output_dataset);
-        node_output.get_value(PGM_def_sym_output_node_id, node_result_id.data(), -1);
-        node_output.get_value(PGM_def_sym_output_node_energized, node_result_energized.data(), 0, -1);
-        node_output.get_value(PGM_def_sym_output_node_u, node_result_u.data(), 0, 1, -1);
-        node_output.get_value(PGM_def_sym_output_node_u_pu, node_result_u_pu.data(), -1);
-        node_output.get_value(PGM_def_sym_output_node_u_angle, node_result_u_angle.data(), -1);
-        CHECK(node_result_id[0] == 0);
-        CHECK(node_result_energized[0] == 1);
-        CHECK(node_result_u[0] == doctest::Approx(50.0));
-        CHECK(node_result_u_pu[0] == doctest::Approx(0.5));
-        CHECK(node_result_u_angle[0] == doctest::Approx(0.0));
-        CHECK(node_result_id[1] == 4);
-        CHECK(node_result_energized[1] == 0);
-        CHECK(node_result_u[1] == doctest::Approx(0.0));
-        CHECK(node_result_u_pu[1] == doctest::Approx(0.0));
-        CHECK(node_result_u_angle[1] == doctest::Approx(0.0));
-    }
+    SUBCASE("Single power flow") {
+        SUBCASE("Simple power flow") {
+            model.calculate(options, single_output_dataset);
+            node_output.get_value(PGM_def_sym_output_node_id, node_result_id.data(), -1);
+            node_output.get_value(PGM_def_sym_output_node_energized, node_result_energized.data(), 0, -1);
+            node_output.get_value(PGM_def_sym_output_node_u, node_result_u.data(), 0, 1, -1);
+            node_output.get_value(PGM_def_sym_output_node_u_pu, node_result_u_pu.data(), -1);
+            node_output.get_value(PGM_def_sym_output_node_u_angle, node_result_u_angle.data(), -1);
+            CHECK(node_result_u[0] == doctest::Approx(50.0));
+            CHECK(node_result_u_pu[0] == doctest::Approx(0.5));
```

Comment:
any particular reason why you check less now?

Reply:
Oh, basically, I combined the checks down below which were repeated. 
Hence all the common checks are below the SUBCASE
I combined the get values too now.

Reply:
oh wait no, i think we had a team agreement to do common checks in a separate function (e.g. a lambda) and then call the function at the end of the subcase, because some people thought the logic flow is more difficult if there are checks after the subcases are exited. please follow that convention